lib/logmsg/logmsg-serialize.c

```diff
@@ -126,17 +166,323 @@ _deserialize_message(LogMessageSerializationState *state)
   return TRUE;
 }
 
+static gboolean
+log_msg_read_tags(LogMessage *self, SerializeArchive *sa)
+{
+  gchar *buf;
+  gsize len;
+
+  while (1)
+    {
+      if (!serialize_read_cstring(sa, &buf, &len) || !buf)
+        return FALSE;
+      if (!buf[0])
+        {
+          /* "" , empty string means: last tag */
+          g_free(buf);
+          break;
+        }
+      log_msg_set_tag_by_name(self, buf);
+      g_free(buf);
+    }
+
+  self->flags |= LF_STATE_OWN_TAGS;
+
+  return TRUE;
+}
+
+/*
+    Read the most common values (HOST, HOST_FROM, PROGRAM, MESSAGE)
+    same for all version < 20
+*/
+gboolean
+log_msg_read_common_values(LogMessage *self, SerializeArchive *sa)
+{
+  gchar *host = NULL;
+  gchar *host_from = NULL;
+  gchar *program = NULL;
+  gchar *message = NULL;
+  gsize stored_len=0;
+  if (!serialize_read_cstring(sa, &host, &stored_len))
+    return FALSE;
+  log_msg_set_value(self, LM_V_HOST, host, stored_len);
+  free(host);
+
+  if (!serialize_read_cstring(sa, &host_from, &stored_len))
+    return FALSE;
+  log_msg_set_value(self, LM_V_HOST_FROM, host_from, stored_len);
+  free(host_from);
+
+  if (!serialize_read_cstring(sa, &program, &stored_len))
+    return FALSE;
+  log_msg_set_value(self, LM_V_PROGRAM, program, stored_len);
+  free(program);
+
+  if (!serialize_read_cstring(sa, &message, &stored_len))
+    return FALSE;
+  log_msg_set_value(self, LM_V_MESSAGE, message, stored_len);
+  free(message);
+  return TRUE;
+}
+
+/*
+    After the matches are read the details of those have to been read
+    this process is same for version < 20
+*/
+gboolean
+log_msg_read_matches_details(LogMessage *self, SerializeArchive *sa)
+{
+  gint i;
+  for (i = 0; i < self->num_matches; i++)
+    {
+      guint8 stored_flags;
+      if (!serialize_read_uint8(sa, &stored_flags))
+        return FALSE;
+      // the old LMM_REF_MATCH value
+      if (stored_flags & 0x0001)
```

Comment:
I do not think I can change the code to reflect about the else branch, only I could use the comment above to produce `OLD_LMM_REF_MATCH` constant instead of the number. I think it improves at least a little.